pmd-apex/src/main/java/net/sourceforge/pmd/lang/apex/rule/bestpractices/ApexUnitTestClassShouldHaveAssertsRule.java

```diff
@@ -93,37 +93,45 @@ private Object checkForAssertStatements(ApexNode<?> node, Object data) {
             statements.addAll(blockStatement.findDescendantsOfType(ASTStatement.class));
             methodCalls.addAll(blockStatement.findDescendantsOfType(ASTMethodCallExpression.class));
         }
-        boolean isAssertFound = false;
-
-        for (final ASTMethodCallExpression methodCallExpression : methodCalls) {
-            if (ASSERT_METHODS.contains(methodCallExpression.getFullMethodName().toLowerCase(Locale.ROOT))) {
-                isAssertFound = true;
-                break;
-            }
-        }
-
-        // If we didn't find assert method invocations the simple way and we have a
-        // configured pattern, try it
+        boolean isAssertFound = isAssertMethodCalled(methodCalls);
+    
         if (!isAssertFound) {
             final String additionalAssertMethodPattern = getProperty(ADDITIONAL_ASSERT_METHOD_PATTERN_DESCRIPTOR);
```

Comment:
move this line to `isAssertMethodPatternMatched` method. I have defined Method defination in review comments.